### DISCUSSION TITLE
Override Binary Type to Generate as File Instead of Blob

### DISCUSSION THREAD

**Discussion by troypoulter** [orval-labs/orval#934] [Q&A] [answered]
I'm working with an OpenAPI spec that involves multipart uploads for binary data. By default, Orval generates these fields with the `Blob` type. However, this presents a challenge for me because the `Blob` type doesn't allow specifying a filename, which is necessary for how I construct my `FormData` payloads for uploads.

**Question:**
Is there a built-in way to configure Orval so that it generates `File` instead of `Blob` for binary types in the API specification?

**Workarounds Considered:**
I am considering using a post-processor script to perform a find-and-replace operation on the generated code, but this seems suboptimal. Any better alternatives or native solutions in Orval would be greatly appreciated.

**Comment by DotNet2Web** [comment, marked as answer]
Really an old post but maybe someone encounters the same issue. 

This is what I had to do to fix this.

In your orval.config.ts file add an override for the contenttype. Include the 'multipart/formdata' to it and your done.

It would be:

`{
...,
...,
"override": {
   "contentType": {
       "include": ['multipart/formdata']
   }
 }
}
`

It still generates File to Blob but you can now send the file instead of having it to convert to blob.
